src/lib/prof-lean/id-tuple.c

```diff
@@ -157,6 +157,14 @@ id_tuple_copy
   memcpy(dest->ids, src->ids, ids_bytes); 
 }
 
+bool
+is_id_tuple_empty
+(
+ id_tuple_t *tuple
+)
+{
+  return (tuple->ids_length == 0);
```

Comment:
I think this should be tuple->length, since ids_length is the slots allocated (with seems to be 7 now), length is for the actual valid ids?

Reply:
I don't understand what is going on with the pull request. I deleted this code.  the length will only be 7 or 8 when we are forming one. in tuples you see, length should match ids_length.

Reply:
I probably read some outdated bits, and now I see it got deleted.  I found out length will match ids_length when I get them, but it seems like 3-5 for gpu and 2-3 for cpu though... how it will only be 7 or 8?

Reply:
This section is deleted in the final version, it should be fine.